tools/BaseRunner.py

```diff
@@ -55,6 +55,16 @@ def run(self, tmp_dir, params):
 
         return (log.decode('utf-8'), proc.returncode)
 
+    def run_with_profiling(self, tmp_dir, params):
+        """Wrapper around run method.
+        Adds user time, system time and ram usage to the tuple returned by run
+        """
+
+        run_result = self.run(tmp_dir, params)
+        usage = resource.getrusage(resource.RUSAGE_CHILDREN)
+        profiling_data = (usage.ru_utime, usage.ru_stime, usage.ru_maxrss)
+        return run_result + profiling_data
```

Comment:
What are the results for the runners that run mostly in python? Do they make sense here? Maybe these tools (I guess only tree-sitter is already merged) should somehow overwrite this method too?